### PR TITLE
Log Viewer or ConsoleLogGUI

### PR DESCRIPTION
**Pull request by lazynessmind**
### Done:

Checking the console output while developing the game is very handy right? With this in mind, I implemented the ability to check the current log output by pressing the `L` key. 

### Notes:

- The `ConsoleLog` can be open from every GUI since it's added as an overlay.
- Currently, you use `L` to show and hide the `ConsoleLog` GUI.
- The ScrollContainer is automatically set to the end. 
- Log levels are available on ### DEBUG, WARN, and ERROR with the respective colors being **white, yellow, and red**.

### Maybe in the future:

- Add a command output to run the in-game commands, or show dev-only features.

### Result: 
image
image

### CODE REVIEW

File: src/cubyz/gui/ConsoleLog.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            try {
+                int lastLogLevel = DEBUG;
+                for (String line : Files.readAllLines(Paths.get(currentLogPath))) {
+                    if (line.contains("| warning |")) {
+                        lastLogLevel = WARN;
+                    } else if (line.contains("| error |")) {
+                        lastLogLevel = ERROR;
+                    } else if (line.contains("| info |") || line.contains("| debug |")) {
+                        lastLogLevel = DEBUG;
+                    }
```
Comment (IntegratedQuantum) [inline src/cubyz/gui/ConsoleLog.java:76]: There is also "| crash |" which unlike the name suggests, doesn't necessarily crash the entire game. Often the "client" is still available.

**Review by IntegratedQuantum — CHANGES_REQUESTED** [review]
It's probably not useful for developers like me, but it is useful for modders and bug reports.

There are a few issues and things you didn't know, but overall it seems good.

In the future this would be more useful in the launcher, but the launcher is temporary anyways, and until it is rewritten in a native language it would be a waste to add more stuff there.

File: src/cubyz/utils/Logger.java [hunk 2]
```diff
@@ -31,7 +33,9 @@
 			}
 			
 			latestLogOutput = new FileOutputStream("logs/latest.log");
-			currentLogOutput = new FileOutputStream("logs/" + logFileFormat.format(Calendar.getInstance().getTime()) + ".log");
+			String currentLogName = "logs/" + logFileFormat.format(Calendar.getInstance().getTime()) + ".log";
+			currentLogOutput = new FileOutputStream(currentLogName);
+			ConsoleLog.setLogFile(currentLogName);
```
[the remark this answers is not in the document]
Comment (lazynessmind) [reply src/cubyz/utils/Logger.java:38]: I tought was the previous log file

File: src/cubyz/gui/ConsoleLog.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+        public LogLabel(String rawText, int color) {
+            this.rawText = rawText;
+            this.color = color;
+        }
+
+        @Override
+        public void render(int x, int y) {
+            Graphics.setColor(color);
+            Graphics.setFont(Fonts.PIXEL_FONT, 16f);
```
[the remark this answers is not in the document]
Comment (lazynessmind) [reply src/cubyz/gui/ConsoleLog.java:106]: My problem with regular labels was that the label itself was using the given string as a translatable key and printing all the log file as not able to translate

File: src/cubyz/utils/Logger.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (IntegratedQuantum) [reply src/cubyz/utils/Logger.java:38]: No. It's always the most recent one, even if the game is still running.

**Comment by lazynessmind** [comment]
All done.

**Pull request merged**